### ISSUE TITLE
Bessere Sortierung

### ISSUE THREAD

**Issue by MaluNoPeleke** [lastship/plugin.video.lastship#15]
Auch wenn die Sortierung bewusst zufällig erfolgt, wäre es doch schön zumindest nach bekannter Qualität zu unterscheiden, sodass HD/SD Material gruppiert aufgelistet wird.
Eine optionale Sortierung nach hochgeladenem Datum könnte die vermeintliche Qualität innerhalb dieser Gruppen dann noch weiter verbessern (jünger ~ bessere Qualität).

**Comment by merlux** [comment]
Die results sollten zumindest nach den verfügbaren Qulitätsangaben der Hoster sortiert sein. Zuerst 1080, dann 720, dann der Rest. 

Ist das bei dir nicht so?

**Comment by MaluNoPeleke** [comment]
Nein, da war zwar ein HD-Stream ganz oben, ein anderer allerdings irgendwo in der Mitte (stable, kein nightly).
Insgesamt wünschte ich mir eine Möglichkeit besser vorhersagen zu können, ob die Qualität einem gewissen Mindeststandard genügt, was aber wohl Wunschdenken bleibt. Man könnte höchstens überlegen eine Zusatzinformation durch xREL zu ermöglichen, die neue Qualitätssprünge aufgrund von besseren Releases ab Datum x neu gruppiert...

**Comment by merlux** [comment]
Es ist leider oft schwer / unmöglich die Qualität eines angebotenen Streams vorherzusagen, da müssen wir uns auf die Qualitätsangaben der Provider verlassen.

xREL listet leider auch nur die Release-Namen, bei lastship bekommen wir allerdings nur die Links zu Hoster (z.B. Openload.co/lajshdflkjashdflkjhasdf.mpv). Da kann man dann nix vergleichen.

Es gäbe die Möglichkeit die Streamqualität über ffmpeg herauszufinden, dafür muss man den Stream allerdings laden und beginnen abzuspielen. Das für einen Film mit >10 Streams zu machen würde lange dauern, da man jeden Stream einzeln resolven und laden müsste.

Ich befürchte da kann man nicht viel machen. Sollten die Qualitäts-infos von Providern verfügbar sein, werden die bereits berücksichtigt und entsprechend sortiert.
Solltest du einen Provider mit Qualitätsangaben auf deren Homepage sehen, welche in Lastship nicht angezeigt werden, sag bitte Bescheid, dann können wird das fixen.

**Comment by MaluNoPeleke** [comment]
Ich meinte auch mehr im Sinne von "seit 12.03. LD" -> alle neuen Uploads ab da in eine neue Gruppe, da höhere Wahrscheinlichkeit einer besseren Qualität.
Natürlich nur wenn das automatisiert machbar wäre

**Comment by merlux** [comment]
Soweit ich weiß haben die wenigsten Hoster eine Info wann welcher Stream von einem Film hochgeladen wurde. Kann mich auch irren.

**Comment by Vikijaka** [comment]
Im bezug auf die Sortierung möchte ich außerdem Anfragen, ob es eventuell möglich wäre "subbed" Streams weiter "unten/hinten" aufzulisten. Begründet durch den Wegfall von BS.to und dem damit verbundenen ausweichen auf Serienstream (welche einiges an Serien in GerDub und O-Ton mit Untertitel anbietet), werden, sofern man Real-Debrid für "Premiumlinks" nutzt, Links mit "Subbed" im Titel, im Auto-Play, priorisiert abgerufen.

**Comment by merlux** [comment]
phu... dafür eventuell ein eigenes Issue mit der erklärung öffnen. Aber auch hier sehe ich das Problem mit der fehlenden Information seitens der stream-anbieter. Da sieht man oft nicht raus, ob es subbed ist.
